packages/angular-output-target/src/generate-angular-component.ts

```diff
@@ -61,7 +75,19 @@ export class ${tagNameAsPascal} {`,
     lines.push(
       `  /** ${output.docs.text} ${output.docs.tags.map((tag) => `@${tag.name} ${tag.text}`)}*/`,
     );
-    lines.push(`  ${output.name}!: I${cmpMeta.componentClassName}['${output.method}'];`);
+    // Regexp to replace output references with their remapped name
+    const outputTypeRemapped = Object.entries(outputReferenceRemap).reduce((type, [src, dst]) => {
+      return type
+        .replace(new RegExp(`^${src}$`, 'g'), `${dst}`)
+        .replace(
+          new RegExp(`([^\\w])${src}([^\\w])`, 'g'),
+          (v, p1, p2) => [p1, dst, p2].join(''),
+        );
+    }, output.complexType.original
+      .replace(/\n/g, ' ')
+      .replace(/\s{2,}/g, ' ')
+      .replace(/,\s*/g, ', '));
```

Comment:
Could you add a few comments to walk me through your process here?

Reply:
Yes, of course: 

The original attribute contains the original type defined by the devs. This could include linebreaks, multiple consecutive spaces, and multiple spaces after commas.

The idea of this is to normalize the original type common issues, by removing linebreaks, replacing consecutive spaces with a single space, and adding a single space after commas.

Do you want me to add more information by adding comments to this part ?

Reply:
Got it! Added your comment above the regexp. 👍